### DISCUSSION TITLE
Pythran and Numba

### DISCUSSION THREAD

**Discussion by rarygit** [pylbm/pylbm#158] [General]
Looking forward to seeing Pythran implemented in pylbm.

Stumbled across this performance comparison published by Augier et al. 2021:

"Reducing the ecological impact of computing through education and Python compilers"
[https://www.nature.com/articles/s41550-021-01342-y.epdf?sharing_token=D6WDBKpfOOcWHERGZz4AbtRgN0jAjWel9jnR3ZoTv0Pc7q-oiK1_CxsdnLnqzBdV9-Xj6CQQ98qvVv9fAgz7ySxWpoPJr5C4XgOXuetbs26fddzM8jDAaP_RUyzuPDdOG_uCV2N67FsOGrLM5IQku9K2si3Kw7cSPET-ES0qGQc%3D](url)

Pythran on 12 cores and 6 cores - impressive 'Time to solution'!

**Comment by gouarin** [comment]
Thanks @rarygit for your message.

I don't think we will have a different execution time if we use Pythran or Numba since the Cython version is already optimized. But indeed, it would be nice to be able to make a comparison!

**Comment by rarygit** [comment]
@gouarin, that's an interesting comment you make regarding Cython optimisation.

I am new to this so cannot go further than the readings I have looked at earlier today.

As to comparisons, it would seem that Pierre Augier's team work on Transonic might be interesting.
The Transonic documentation has a page dedicated to this topic:  "Writing Benchmarks"
[https://transonic.readthedocs.io/en/latest/examples/writing_benchmarks/bench.html](url)

> "Transonic is a pure Python package (requiring Python >= 3.6) to easily accelerate modern Python-Numpy code with different accelerators (currently Cython, Pythran and Numba, but potentially later Cupy, PyTorch, JAX, Weld, Pyccel, Uarray, etc...).
The accelerators are not hard dependencies of Transonic: Python codes using Transonic run fine without any accelerators installed (of course without speedup)"
[https://github.com/fluiddyn/transonic](url)
